app/models/submission.rb

```diff
@@ -122,6 +139,25 @@ def self.zip_submissions!(tutorial, assignment)
     archived_filestream
   end
 
+  def self.zip_corrections!(tutorial, assignment)
+    submissions = Submission.where(tutorial: tutorial,
+                                   assignment: assignment).proper
+    begin
+      archived_filestream = Zip::OutputStream.write_buffer do |stream|
+        submissions.each do |s|
+          if s.correction
+            stream.put_next_entry(s.filename_for_corrections_bulk_download)
+            stream.write IO.read(s.correction_file_path)
+          end
+        end
+      end
+      archived_filestream.rewind
+    rescue => e
+      archived_filestream = e.message
+    end
+    archived_filestream
+  end
+
```

Comment:
Same as above: It should be possible to refactor `zip_submissions!` and `zip_corrections!` in order to avoid code duplication.